pyenv-win/bin/pyenv.bat

```diff
@@ -46,61 +46,97 @@ if /i not [%1]==[exec] goto :plugin
 :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 :exec
 
-if not exist "%bindir%" (
+if not exist "%__pyenv_bindir%" (
   echo No global python version has been set yet. Please set the global version by typing:
   echo pyenv global 3.7.2
   exit /b
 )
 
-set cmdline=%*
-set cmdline=%cmdline:~5%
+set "__pyenv_cmdline=%*"
+set "__pyenv_cmdline=%__pyenv_cmdline:~5%"
 :: update PATH to active version and run command
 :: endlocal needed only if cmdline sets a variable: SET FOO=BAR
-set cmddir=
-if exist %bindir%\%2 set cmddir=%bindir%\
-if exist %bindir%\%2.exe set cmddir=%bindir%\
-if exist %bindir%\%2.bat set cmddir=%bindir%\
-set "path=%extrapaths%%path%"
-%cmddir%%cmdline%
+call :remove_shims_from_path
+%__pyenv_cmdline%
 endlocal
 exit /b
 :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+:remove_shims_from_path
+:: arcane magic courtesy of StackOverflow question 5471556
```

Comment:
```suggestion
:: arcane magic courtesy of StackOverflow question 5471556
:: https://stackoverflow.com/a/5472168/305023
```

Reply:
What about problems raised by https://stackoverflow.com/a/7940444/305023 ?

Reply:
I didn't look beyond the accepted answer. I switched over to that one.